### DISCUSSION TITLE
set_location() doesn't work

### DISCUSSION THREAD

**Discussion by gbcbooks** [appium/appium#17732] [Q&A]
hello everyone

i try to mock lcation to wechat, but it seems doesn't work, i was wondering is my code something wrong ? 

my CAPS

```
EXECUTOR = 'http://127.0.0.1:4723/wd/hub'

ANDROID_CAPS = {
                "platformName": "Android",
                "platformVersion": "7.0",
                "udid": SN,
                "deviceName": "SUMSONG_S6" ,
                "automationName": "uiautomator2",
                # "fastReset": "false",
                "autoGrantPermissions": "true",
                "newCommandTimeout": 10000,
                "unicodeKeyboard": True,
                "resetKeyboard": True,
                "noReset" : True,
                "gpsEnabled": True,
                "recreateChromeDriverSessions": True,
                "chromeOptions": {"androidProcess": "com.tencent.mm:appbrand0", "androidProcess": "com.tencent.mm:tools"}
            }
```
python log
```
driver = startDriver(EXECUTOR, ANDROID_CAPS)
driver.set_location(233,452,2)
driver.start_activity("com.tencent.mm",".ui.LauncherUI")
```
after above ,  i operate wechat to send a location , but , mock location doesn't work.

**Comment by KazuCocoa** [comment]
What was the appium server log?
It might take a few seconds to update the location because of Android OS side's thing. So potentially it does not work just after changing the location. https://github.com/appium/appium-uiautomator2-driver#mobile-refreshgpscache may help to apply the change forcefully. (It has a fix with latest UIA2 driver, available over Appium 2)
